# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/src/XPCJSContext.cpp
# Commit: 4880b1f5b3bc
# Full Hash: 4880b1f5b3bc4f9f03c976f7af94329e8004cd6e
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2023-12-06 05:18:37
# Regressor Bug: 1589554
# File Overlap Count: 1
# Description:
#   Bug 1589554 - Part 4: Release/Deny screen-wake-lock for implementation-specific reasons. r=dom-core,edgar
#   
#   Depends on D189510
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189511
# ==============================================================================

diff -r f932a2a98670 -r 4880b1f5b3bc js/xpconnect/src/XPCJSContext.cpp
--- a/js/xpconnect/src/XPCJSContext.cpp	Tue Dec 05 23:58:06 2023 +0000
+++ b/js/xpconnect/src/XPCJSContext.cpp	Tue Dec 05 23:58:07 2023 +0000
@@ -48,9 +48,11 @@
 #include "js/WasmFeatures.h"
 #include "mozilla/dom/BindingUtils.h"
 #include "mozilla/dom/ContentChild.h"
+#include "mozilla/dom/Document.h"
 #include "mozilla/dom/Element.h"
 #include "mozilla/dom/ScriptLoader.h"
 #include "mozilla/dom/WindowBinding.h"
+#include "mozilla/dom/WakeLockBinding.h"
 #include "mozilla/extensions/WebExtensionPolicy.h"
 #include "mozilla/Atomics.h"
 #include "mozilla/Attributes.h"
@@ -747,6 +749,9 @@
     if (Preferences::GetBool("dom.global_stop_script", true)) {
       xpc::Scriptability::Get(global).Block();
     }
+    if (nsCOMPtr<Document> doc = win->GetExtantDoc()) {
+      doc->UnlockAllWakeLocks(WakeLockType::Screen);
+    }
     return false;
   }
 
